adios2-config cannot be used on Mac?

Even though adios2-config gives the link flag, it does not work for shared libraries on a Mac like it does on Linux:

adiosvm:Tutorial/heat2d/fortran

pnb13:fortran pnb$ echo $DYLD_LIBRARY_PATH 
/opt/adios2/lib:/opt/chaos/lib:

pnb13:fortran pnb$ /opt/adios2/bin/adios2-config --fortran-libs
/opt/adios2/lib/libadios2_f.dylib 
pnb13:fortran pnb$ make adios2
cd simulation; make adios2
mpif90 -g -O2 -o heatSimulation_adios2 `/opt/adios2/bin/adios2-config --fortran-flags` heat_vars.o io_adios2.o heatSimulation.F90 `/opt/adios2/bin/adios2-config --fortran-libs` 
ld: file not found: libadios2.2.dylib for architecture x86_64
collect2: error: ld returned 1 exit status
make[1]: *** [adios2] Error 1
make: *** [adios2] Error 2

I still have to manually point to the dependent libraries. 
How should this be used on a Mac?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,11 @@ set(CMAKE_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/adios2
   CACHE STRING "Installation CMake subdirectory")
 mark_as_advanced(CMAKE_INSTALL_CMAKEDIR)
 
+# On Mac OS X, set the dir. included as part of the installed library's path:
+if (BUILD_SHARED_LIBS AND NOT DEFINED CMAKE_INSTALL_NAME_DIR)
+  set(CMAKE_INSTALL_NAME_DIR "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}")
+endif ()
+
 list(INSERT CMAKE_MODULE_PATH 0 "${PROJECT_SOURCE_DIR}/cmake")
 if(NOT CMAKE_ARCHIVE_OUTPUT_DIRECTORY)
   set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY
@@ -95,7 +100,7 @@ endif()
 
 include(CMakeDependentOption)
 
-# Setup shared library defaults.  If explicitly specified somehow, then default 
+# Setup shared library defaults.  If explicitly specified somehow, then default
 # to that.  Otherwise base the default on whether or not shared libs are even
 # supported.
 get_property(SHARED_LIBS_SUPPORTED GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS)

